.github/workflows/test-rust.yaml

```diff
@@ -0,0 +1,75 @@
+name: test-rust
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - ".github/workflows/test-rust.yaml"
+  workflow_call:
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+          - os: windows-latest
+          - os: macos-latest
+          - os: ubuntu-latest
+            target_option: "--target=wasm32-unknown-unknown"
```

Comment:
For PRs, we want these to run really quickly, so we only run on ubuntu, which catches all but the rarest errors (maybe 1/1000 errors are only on windows). Then on `test-all` we want to run all the OSs.

So I think it might be worth going back to defining this list of OSs in `test-all` and passing the values to `test-rust`.
`pull-request` already defines the two options it wants for `target_option`, so that should work.

Does that make sense? Lmk if it's confusing...